python/write_file.py

```diff
@@ -0,0 +1,27 @@
+"""Script to write text to a new file.
+Run script as `python write_file.py <output_file>.py "some text"`
+"""
+import sys
+import os
+
+if len(sys.argv) != 3:
+    print("Don't forget to enter the output file path and a string")
+
+outfile = sys.argv[1]
+text = sys.argv[2]
+
+if os.path.exists(outfile):
+   print("File already exists, clean this up to be sure we're writing/reading a new file")
```

Comment:
I normally would raise errors here but not sure what happens to the errors in the docker container and if we'd ever see them, or everything would grind to a halt. Any thoughts as to if I should be raising errors, or if that is going to be annoying in other languages?

Reply:
Errors will only come out as text written to stderr (or sometimes to stdout) or as the process' exit code being non-zero. So, if an error is raised, at worst we'd know because the test fails and then we'd dig in to figure out what's going on.

I think ultimately, it might be nice to set up some sort of hook in pytest that, when a test fails, it prints all the stdout, stderr, exit code from the container. That would likely save a bunch of time in figuring out why a test is failing.

In this case specifically, if we want to ensure the test environment is set up right (i.e. not having that file already there), that shouldn't be the job of this script. Either the Dockerfile should perform the sanity check, or, if we end up using `docker exec` to test for the existence of and contents of the file, we could also use it to check for the *non*-existence of the file before we run the script.

Reply:
Actually, instead of `docker exec`, we can put it the docker run command, like this:

```python
@pytest.mark.parametrize(
    'docker_container',
    ["""
      if [ -f output.txt ]; then
        echo 'Error: output.txt already exists'
        exit 1
      fi

      python write_file.py output.txt "Bob Barker"
      cat output.txt
    """],
```

Reply:
I can't figure out how to pass in the bash lines to the docker command.
This works:
```
    @pytest.mark.parametrize(
        'docker_container',
        [['/bin/sh', '-c', 'python write_file.py output.txt "Bob Barker"; cat output.txt']],
        indirect=True,
    )
```
but not
```
    @pytest.mark.parametrize(
        'docker_container',
        ['python write_file.py output.txt "Bob Barker"; cat output.txt'],
        indirect=True,
    )
```
and I still can't grasp why.

Reply:
Sorry, I confused myself and maybe you, too. Here's a comment where I point out where I got confused: https://github.com/Kinnich/rosetta-io/pull/11/files#r1298866911

Therefore, we'll have to use the `[['/bin/sh', '-c', ...]]` approach when we want to run shell commands (like using the `...; cat output.txt` trick or when we want to redirect a file to a script's stdin). We can simplify it as `[['sh', '-c', ...]]]`

The problem was that somewhere along the line, our string gets turned into an array in order to be helpful

e.g. this:
`['python write_file.py output.txt "Bob Barker"; cat output.txt'],`

gets turned into this:
`[['python', 'write_file.py', 'output.txt', 'Bob Barker;', 'cat', 'output.txt']],`

So, `cat` is now the 4th argument to the script, instead of another command